core/MyProtocolMySensors.cpp

```diff
@@ -120,40 +120,42 @@ bool protocolMQTTParse(MyMessage &message, char* topic, uint8_t* payload, unsign
 	uint8_t bvalue[MAX_PAYLOAD];
 	uint8_t blen = 0;
 	uint8_t command = 0;
+	uint8_t topiclen = strlen(topic) - strlen(MY_MQTT_SUBSCRIBE_TOPIC_PREFIX);
+	char topicTrimmed[topiclen];
 
-	for (str = strtok_r(topic, "/", &p); str && i <= 5;
+  if (topic != strstr(topic, MY_MQTT_SUBSCRIBE_TOPIC_PREFIX)) {
+		// Prefix doesn't match incoming topic
+		return false;
+	}
+	// Extract the topic sans prefix
+	strncpy(topicTrimmed, &topic[strlen(MY_MQTT_SUBSCRIBE_TOPIC_PREFIX) + 1], topiclen);
+  topicTrimmed[topiclen] = (char)0;
+
+	for (str = strtok_r(topicTrimmed, "/", &p); str && i <= 5;
```

Comment:
Do you really need to copy topic into the new buffer topicTrimmed?

I _think_ you just could add to the topic pointer this:

for (str = strtok_r(topic + strlen(MY_MQTT_SUBSCRIBE_TOPIC_PREFIX), "/", &p);

Maybe you could test this in your environment?